SAP/open-ux-tools
# feat: introduce app-config-writer with smartlinks-config

#916 

### Description
Introduce app-config-writer to support smart link configuration in Fiori apps
- Adds SmartLinks configuration to local project 
  - Find smartlinks configuration target from `ui5-deploy.yaml/ui5.yaml`
  - Add pre-configured targets for the included smart links

### Changes
- `@sap-ux/project-access`: introduce `readUi5Yaml` to get contents of Ui5 yaml file
- `@sap-ux/ui5-config`: export and extend types, add `FioriToolsServeStaticConfig` middleware type
- `@sap-ux/add-config-writer`: introduce app-config writer to add smartlinks config
- `@sap-ux/create`: add `add smartlinks-config` command

### How to test
In test project, install:
- [sap-ux-app-config-writer-0.0.1.tgz](https://github.com/SAP/open-ux-tools/files/11050667/sap-ux-app-config-writer-0.0.1.tgz)
- [sap-ux-create-0.0.12.tgz](https://github.com/SAP/open-ux-tools/files/10981314/sap-ux-create-0.0.12.tgz)
- [sap-ux-project-access-1.2.0.tgz](https://github.com/SAP/open-ux-tools/files/10981312/sap-ux-project-access-1.2.0.tgz)

Then you should be able to run `npx @sap-ux/create add smartlinks-config`
